src/Sarif.TestUtilities/DeletesOutputsDirectoryOnClassInitializationFixture.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif.TestUtilities
+{
+    /// <summary>
+    /// This class is invoked a single time on initializing an xunit class for testing. It deletes
+    /// any existing test output files that may have been produced by the previous run. This fixture
+    /// is required because individual tests in test classes result in an object instantiation for
+    /// each test case. The FileDiffingTests pattern, by design outputs a bundle of outputs to 
```

Comment:
>FileDiffingTests [](start = 28, length = 16)

Since this fixture is intended to support classes that derive from `FileDiffingTests`, why not do this:
```
class FileDiffingTests: IClassFixture<DeletesOutputsDirectoryOnClassInitializationFixture>
{
    ...
}

class FortifyFprConverterTests: FileDiffingTests
{
    ...
}
```

... rather than marking each derived class as implementing `IClassFixture`. Does XUnit's reflection not find the interface if it's on a base class?

Reply:
This might relate to the comment I made on `FortifyFprConverterTests`, where I asked why you bothered to derive a class from `DeletesOutputsDirectoryOnClassInitializationFixture`. It's true that if you want to leave the flexibility for each derived test class to have its own fixture, then you can't put the marker interface on the base class.

If that was your reasoning, it's worth a comment here.

---
In reply to: [247725332](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247725332) [](ancestors = 247725332)

Reply:
Ok, I see now that `DeletesOutputsDirectoryOnClassInitializationFixture` has some virtual properties that derived classes can override, so you really do intend each derived class to have a shot at it. I've been reworking this comment, so I'll mention that.

---
In reply to: [247725737](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247725737) [](ancestors = 247725737,247725332)

Reply:
Ok, I updated this comment with my accustomed crystalline authorial voice :-)

---
In reply to: [247729832](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247729832) [](ancestors = 247729832,247725737,247725332)